cgi-bin/DW/Controller/Customize.pm

```diff
@@ -0,0 +1,654 @@
+            {
+            url   => LJ::create_url("/customize/advanced/"),
+            title => LJ::Lang::ml('widget.currenttheme.options.advancedcust')
+            };
+        push @current_options,
+            {
+            url =>
+                LJ::create_url( "/customize/advanced/", args => { id => $theme_data->{layoutid} } ),
+            title => LJ::Lang::ml('widget.currenttheme.options.editlayoutlayer')
+            }
+            if $theme_data->{can_edit_layout};
+        push @current_options,
+            {
+            url =>
+                LJ::create_url( "/customize/advanced/", args => { id => $theme_data->{themeid} } ),
+            title => LJ::Lang::ml('widget.currenttheme.options.editthemelayer')
+            }
+            if $theme_data->{can_edit_theme};
+    }
+
+    push @current_options,
+        { url => "#layout", title => LJ::Lang::ml('widget.currenttheme.options.layout') };
+
+    return { 'current' => $theme_data, 'current_options' => \@current_options };
+}
+
+sub get_layout_data {
+    my $u = LJ::get_effective_remote();
+    die "Invalid user." unless LJ::isu($u);
+
+    my $current           = LJ::Customize->get_current_theme($u);
+    my %layouts           = $current->layouts;
+    my $show_sidebar_prop = $current->show_sidebar_prop;
+    my %layout_names      = LJ::Customize->get_layouts;
+
+    my $layout_prop = $current->layout_prop;
+
+    my $prop_value;
+    if ( $layout_prop || $show_sidebar_prop ) {
+        my $style = LJ::S2::load_style( $u->prop('s2_style') );
+        die "Style not found." unless $style && $style->{userid} == $u->id;
+
+        if ($layout_prop) {
+            my %prop_values = LJ::Customize->get_s2_prop_values( $layout_prop, $u, $style );
+            $prop_value = $prop_values{override};
+        }
+
+        # for layouts that have a separate prop that turns off the sidebar, use the value of that
+        # prop instead if the sidebar is set to be off (false/0).
+        if ($show_sidebar_prop) {
+            my %prop_values = LJ::Customize->get_s2_prop_values( $show_sidebar_prop, $u, $style );
+            $prop_value = $prop_values{override} if $prop_values{override} == 0;
+        }
+    }
+
+    my @layout_data = ();
+    for my $layout ( sort keys %layouts ) {
+        my $tmp = {
+            layout => $layout,
+            name   => $layout_names{$layout},
+        };
+        $tmp->{current} =
+            ( !$layout_prop ) || ( $layout_prop && $layouts{$layout} eq $prop_value ) ? 1 : 0;
+
+        push @layout_data, $tmp;
+    }
+
+    my $is_system = $current->is_system_layout;
+
+    return {
+        is_system         => $current->is_system_layout,
+        show_sidebar_prop => $show_sidebar_prop,
+        layout_prop       => $layout_prop,
+        layouts           => \@layout_data
+    };
+
+}
+
+sub make_queryargs {
+    my $args = shift;
+    my $ret_args;
+
+    $ret_args->{cat}      = defined $args->{cat}      ? $args->{cat}      : "";
```

Comment:
nit: all of these can be replaced with the form,

```perl
$ret_args->{cat} //= "";
```

All hail the defined-or operator!